### PR TITLE
DHFPROD-9535: Error creating a concept with existing

### PR DESCRIPTION
**Pull request by iveyMarklogic**
#### Checklist: 
```diff
- Note: do not change the below
```

- ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
- [x] Ran cypress tests on firefox locally
  

- ##### Reviewer:

- [x] Reviewed Tests
- [n/a] Added to Release Wiki

### CODE REVIEW

**Review by Sanjeevani19 — COMMENTED** [review]
@iveyMarklogic  Can we add a test to cover this scenario ?

**Comment by iveyMarklogic** [comment]
@Sanjeevani19  I added the unit tests to cover this scenario

**Review by briantang — DISMISSED** [review]
Looks good!

**Review by briantang — DISMISSED** [review]

**Review by littlecherro — CHANGES_REQUESTED** [review]
I think its a minor bug, but i tried to create a concept class with the same name as an entity and this text appear. Member is an Entity NOT a concept class :) The same happens if i try to create an entity with a concept class name that is being used.

image

**Comment by littlecherro** [comment]
i also think that we should use onBlur insead of making the text appear after clicking "Add", what do you think?

**Comment by iveyMarklogic** [comment]
@littlecherro  both scenarios work as expected. First, the model does not allow you to create an element with an existing name in the graph, regardless of whether the name was used in an entity or in a concept. second is expected the validation runs when you hit the add button

**Comment by littlecherro** [comment]
> @littlecherro both scenarios work as expected. First, the model does not allow you to create an element with an existing name in the graph, regardless of whether the name was used in an entity or in a concept. second is expected the validation runs when you hit the add button

@iveyMarklogic 

Ok for the onBlur part.

For the first comment i left, what i meant was that the text is kind of confusing. Its saying that a concept class already exist with that name, but its an Entity that has that name. Also if we want to state that a concept/entity cannot have a repeated name from any "class" (concept or entity), the text should be different, just that :)

**Review by littlecherro — APPROVED** [review]
Message should be updated, and it will be changed in another ticket

File: marklogic-data-hub-central/ui/src/components/modeling/concept-class-modal/concept-class-modal.tsx [hunk 1]
```diff
@@ -22,6 +22,7 @@ type Props = {
   updateConceptClassAndHideModal: (conceptClass: string, description: string) => void;
   updateHubCentralConfig: (hubCentralConfig: any) => void;
   hubCentralConfig: hubCentralConfig;
+  dataModel: Array<any>;
```
Comment (aniamy) [inline marklogic-data-hub-central/ui/src/components/modeling/concept-class-modal/concept-class-modal.tsx:25]: Any reason we cannot add this to the const of props below?

Comment (iveyMarklogic) [reply marklogic-data-hub-central/ui/src/components/modeling/concept-class-modal/concept-class-modal.tsx:25]: no reason just used it like that but I already changed it

**Review by aniamy — APPROVED** [review]

**Review by briantang — APPROVED** [review]

**Pull request merged**